app/src/main/java/com/capstone/weatherapp/CityListAdapter.kt

```diff
@@ -0,0 +1,34 @@
+package com.capstone.weatherapp
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+
+class CityListAdapter (private val cityList : List<Int>) : RecyclerView.Adapter<CityListAdapter.ViewHolder>() {
+    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        val nameTextView = itemView.findViewById<TextView>(R.id.city_name)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CityListAdapter.ViewHolder {
+        val context = parent.context
+        val inflater = LayoutInflater.from(context)
+
+        val cityView = inflater.inflate(R.layout.item_city, parent, false)
+
+        return ViewHolder(cityView)
+    }
+
+    override fun onBindViewHolder(holder: CityListAdapter.ViewHolder, position: Int) {
+        val city: Int = cityList[position]
+
+        val textView = holder.nameTextView
+        val text = "List Item ${city}"
+        textView.text = text
```

Comment:
there's nothing wrong with the above 3 lines, but it may be cleaner to combine them into one line. Something like this: `holder.nameTextView.text = "List Item ${city}"`

Reply:
for some reason my IDE didn’t like that, probably some standard rule